plugins/linux-capture/xcomposite-helpers.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <obs-module.h>
+#include <X11/Xlib.h>
+#include <xcb/xcb.h>
+#include <glad/glad.h>
+#include <glad/glad_glx.h>
+
+#include <util/dstr.h>
+#include <util/darray.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+void xcomp_gather_atoms(xcb_connection_t *conn);
+bool xcomp_check_ewmh(xcb_connection_t *conn, xcb_window_t root);
+struct darray xcomp_top_level_windows(xcb_connection_t *conn);
+struct dstr xcomp_window_class(xcb_connection_t *conn, xcb_window_t win);
+struct dstr xcomp_window_name(xcb_connection_t *conn, Display *disp,
+			      xcb_window_t win);
+xcb_window_t xcomp_find_window(xcb_connection_t *conn, Display *disp,
+			       const char *str);
+bool xcomp_window_exists(xcb_connection_t *conn, xcb_window_t win);
+void xcomp_cleanup_pixmap(Display *disp, struct xcompcap *s);
+void xcomp_create_pixmap(xcb_connection_t *conn, Display *disp,
+			 struct xcompcap *s, int log_level);
+
+void watcher_register(xcb_connection_t *conn, struct xcompcap *s);
+void watcher_unregister(xcb_connection_t *conn, struct xcompcap *s);
```

Comment:
Could these functions receive `xcb_window_t win` instead of `struct xcompcap`?

Reply:
No they map sources to the window they capture and must know this mapping.